client/state/data-layer/wpcom-http/README.md

```diff
@@ -286,6 +286,25 @@ export default {
 };
 ```
 
+#### "Stateless" handlers
+
+There's something special about many of our handlers that we write: they don't depend on the response.
+One of the most common types of responders is the `announceFailure()` handler which does nothing more dispatch a single Redux action.
+Creating multiple functions just to return a static object can be cumbersome.
+Let's take advantage of a shortcut designed just for these situations: we can choose to send in a disptchable object in place of the handler functions.
+
+```js
+export default {
+	[ REQUEST_LIKES ]: [ dispatchRequest(
+		fetchLikes,
+		updateLikes,
+		createNotice( translate( 'Could not fetch likes, try again.' ) ), // <- this is a simple action
```

Comment:
I like the simplification here. Are we ok here if someone changes their language? I think we still do a full page refresh, but wasn't sure.

Reply:
that's my hope, because if we can forego this one constraint it will make so many cases much easier

Reply:
> Are we ok here if someone changes their language? I think we still do a full page refresh, but wasn't sure.

We do a full page refresh. I explored the question "do we really need to refresh?" in #13712 as a part of my trial project. TLDR is that it's avoidable and the biggest problem is switching from LTR to RTL languages and swapping `style.css` and `style-rtl.css` stylesheets.

Across Calypso, effort is being made to make sure that `translate` is always called "just in time", on every render, with the latest l10n data. React components use the `localize` HOC that forces a rerender on locale change. "Static" localized strings are implemented with something like this:
```js
// don't do this at global module level
const noticeTexts = { in: i18n.translate( 'Hello' ), out: i18n.translate( 'Goodbye' ) };
// do this instead
const noticeTexts = translate => ( { in: translate( 'Hello' ), out: translate( 'Goodbye' ) } );
```

In this sense, static `createNotice( translate( ... ) )` is a regression, although it doesn't immediately break anything.

One solution is to avoid static global actions and do a trivial action creator instead:
```js
dispatchRequest( ..., () => createNotice( i18n.translate( 'Error' ) ) );
```

Now, when a locale gets switched, new notices use the new language, but the old ones, already displayed, use the old language.

I don't know right now what's the perfect solution. I'm afraid that trying to do UI stuff in a data layer handler will always be a bit challenging.

Reply:
This probably needs to pair with changes to the data layer API if we are going to make them return actions, in which case it gets a bit easier.

> I'm afraid that trying to do UI stuff in a data layer handler will always be a bit challenging.

You've raised this issue before but I'd argue there's a difference between doing UI stuff in the data layer and raising global alerts through the notices. In almost every case it's because some network activity failed and by keeping the error-handling out of React we're keeping a considerably simpler model (so many `_FAILURE` Redux actions fly by without even having a single reducer respond let alone UI) and doing a very pragmatic thing: inform the customer that something failed and they might want to try again. If it were actually coupling into the app more tightly and weren't just a basic string I'd be all in with you on that assessment.

Reply:
Noting that this won't be a problem if it happens after changes to the data layer API found in #18733 - in that case there will be no distinction between functions returning actions in general and data layer handlers

that is, after that change we should just be able to send…

```js
dispatchRequest( {
	onFailure: () => createNotice( translate( 'Error' ) )
} )
```